host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1092,11 +1092,11 @@ static FILE *open_filename(RASPIVID_STATE *pState, char *filename)
 
 	  //If %d/%u or any valid combination e.g. %04d is specified, assume segment number.
 	  bool bSegmentNumber = false;
-	  const char* pPercent = strstr(filename, "%");
+	  const char* pPercent = strchr(filename, "%");
```

Comment:
Please retest. Needs to be single quotes around the %, not double (single char vs a string)